swaylock/seat.c

```diff
@@ -85,15 +63,9 @@ static void keyboard_modifiers(void *data, struct wl_keyboard *wl_keyboard,
 	struct swaylock_state *state = data;
 	xkb_state_update_mask(state->xkb.state,
 			mods_depressed, mods_latched, mods_locked, 0, 0, group);
-	xkb_mod_mask_t mask = xkb_state_serialize_mods(state->xkb.state,
-			XKB_STATE_MODS_DEPRESSED | XKB_STATE_MODS_LATCHED);
-	state->xkb.modifiers = 0;
 	state->xkb.caps_lock = xkb_state_mod_name_is_active(state->xkb.state, XKB_MOD_NAME_CAPS, XKB_STATE_MODS_LOCKED);
-	for (uint32_t i = 0; i < MASK_LAST; ++i) {
-		if (mask & state->xkb.masks[i]) {
-			state->xkb.modifiers |= XKB_MODS[i];
-		}
-	}
+	state->xkb.control = xkb_state_mod_name_is_active(state->xkb.state, XKB_MOD_NAME_CTRL, XKB_STATE_MODS_DEPRESSED | XKB_STATE_MODS_LATCHED);
```

Comment:
80 col limit

Reply:
crap, I actually thought of it when I wrote this because the previous line was way too long and forgot to wrap it. done in a force push.